src/test/java/teammates/test/cases/browsertests/InstructorCourseEditPageUiTest.java

```diff
@@ -160,6 +161,23 @@ private void testInviteInstructorAction() {
                 Const.StatusMessages.COURSE_REMINDER_SENT_TO + "InsCrsEdit.newInstr@gmail.tmt");
     }
 
+    private void testCancelAddInstructor() {
+        ______TS("Click cancel button upon new form created");
+        courseEditPage.clickShowNewInstructorFormButton();
+        assertTrue(courseEditPage.verifyAddInstructorFormDisplayed());
+        courseEditPage.clickCancelAddInstructorLink();
+        assertFalse(courseEditPage.verifyAddInstructorFormDisplayed());
+
+        ______TS("Click cancel button after filling some fields of the form");
+        courseEditPage.clickShowNewInstructorFormButton();
+        assertTrue(courseEditPage.verifyAddInstructorFormDisplayed());
+
+        courseEditPage.fillNewInstructorName("New Instructor Test");
+        courseEditPage.fillNewInstructorEmail("newinstructortest@example.com");
```

Comment:
Are these fields being cleared away after we discard? If they are should we test that here?

Specifically, when we do `add instructor` after this. Are those fields pre-filled? Or should they be empty.

Reply:
The fields would not be cleared after the instructor clicks `Cancel`. They will be cleared only after a page reload.

I'll include two more tests:
1. Check if fields remain after pressing cancel button
2. Check if fields are emptied after a page reload

Reply:
Do you think we should clear the fields instead?

Reply:
I feel both approaches have its advantages. Not clearing the field would benefit users who change their mind after cancelling the form. Clearing the fields would seem more intuitive for users wanting to reset the form.
 
I'm fine with either ways 😄

Reply:
@tanhengyeow, both sounds okay to me. But I think that we can try to ensure that the user experience is consistent here. I can't recall from the top of my head where there are other `discard` mechanism in TEAMMATES, but you can match the UX similarly.

Reply:
@Crphang I'll adopt the approach for clearing the fields. The current discard mechanism for cancelling the process of editing instructor details resets the form to its default values. 

Therefore, clearing the fields (similar to resetting the form its default values after cancelling) would allow a more consistent UX :)